src/codegen/visibility.rs

```diff
@@ -0,0 +1,75 @@
+use std::fmt;
+use std::str::FromStr;
+
+#[derive(Debug, Clone, Copy, Eq, PartialEq)]
+pub enum Visibility {
+    Public,
+    Crate,
+    Mod,
+    Super,
+    Private,
+}
+
+impl Visibility {
+    pub fn is_public(&self) -> bool {
+        matches!(self, Self::Public)
+    }
+
+    pub fn export_visibility(&self) -> String {
+        match self {
+            Self::Public => "pub",
+            Self::Private => "",
+            Self::Crate => "pub(crate)",
+            Self::Super => "pub(super)",
+            Self::Mod => "pub(mod)",
+        }
+        .to_owned()
+    }
+}
+
+impl Default for Visibility {
+    fn default() -> Self {
+        Self::Public
+    }
+}
+
+impl fmt::Display for Visibility {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Self::Public => write!(f, "pub"),
+            Self::Crate => write!(f, "pub(crate)"),
+            Self::Mod => write!(f, "mod"),
```

Comment:
This one seems weird, where is it coming from? (Or, in which context does this come up?)

Was this maybe meant to be the entry for `pub(self)`?

Reply:
Ah I actually don't think we want to have a "mod" visibility modifier here as I don't see where it could be used

Reply:
Yes that seems useless